src/DataObjectImporter.php

```diff
@@ -230,22 +260,107 @@ protected function updateThumbnail(MediaInterface $media): void {
       $file->setPermanent();
       $file->save();
     }
-    $media->thumbnail->target_id = $file->id();
 
-    // Set the thumbnail alt text, if available.
+    return $file;
+  }
+
+  /**
+   * Reference the thumbnail using as an image media entity.
+   *
+   * As defined by the source plugin for the given media.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   Mpx video media entity.
+   * @param \Drupal\file\FileInterface $file
+   *   Thumbnail file entity.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function referenceThumbnailAsMedia(MediaInterface $media, FileInterface $file) {
+    $source = $media->getSource();
+    $source_configuration = $source->getConfiguration();
+
+    // Save a media entity for the thumbnail image.
+    $media_image = Media::create([
```

Comment:
I think we should follow what we do for files, and first see if an existing media entity exists. Otherwise, if multiple mpx assets reference the same upstream thumbnail, we could end up with duplicate media entities.